JSONRPC/JSONRPC.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.IO;
+using System.Text;
+
+using Newtonsoft.Json.Linq;
+
+public class JSONRPC {
+	private string _endpoint;
+
+	// Basic authentication
+	private string _username;
+	private string _password;
+
+	public void setEndpoint(string endpoint, string username = null, string password = null) {
+		_endpoint = endpoint;
+		_username = username;
+		_password = password;
+	}
+
+	public void call(string methodName, JObject parameters, Dictionary<string, string> headers, Action<Exception, JObject> cb) {
+		// Make sure the endpoint is set
+		if (string.IsNullOrEmpty(_endpoint)) {
+			cb(new Exception("Endpoint has not been set"), null);
+			return;
+		}
+
+		// Setup JSON request object
+		JObject requestObject = new JObject ();
+		requestObject.Add("id", new JValue(1));
```

Comment:
@ronkorving ids in JSONRPC are only used when sending batches
otherwise the request is always the first in the batch